src/citrine/ara/variables.py

```diff
@@ -604,3 +604,45 @@ def __init__(self, *,
         self.ingredient_name = ingredient_name
         self.quantity_dimension = quantity_dimension
         self.process_templates = process_templates
+
+
+class XOR(Serializable['XOR'], Variable):
+    """[ALPHA] Logical exclusive OR for GEM table variables.
+
+    This variable combines the results of 2 or more variables into a single variable according to
+    exclusive OR logic. XOR is defined when exactly one of its inputs is defined. Otherwise it is
+    undefined.
+
+    XOR can only operate on inputs with the same output type. For example, you may XOR
+    :class:`~citrine.ara.variables.RootIdentifier` with
+    :class:`~citrine.ara.variables.IngredientIdentifierByProcessTemplateAndName` because they both
+    produce simple strings, but not with :class:`~citrine.ara.variables.IngredientQuantityInOutput`
+    which produces a real numeric quantity.
+
+    The input variables to XOR need not exist elsewhere in the table definition, and the name and
+    headers assigned to them have no bearing on how the table is constructed. That they are
+    required at all is simply a limitation of the current API.
+
+    Parameters
+    ---------
+    name: str
+        a short human-readable name to use when referencing the variable
+    headers: list[str]
+        sequence of column headers
+    variables: list[Variable]
+        set of 2 or more Variables to XOR
+
+    """
+
+    name = properties.String('name')
+    headers = properties.List(properties.String, 'headers')
+    variables = properties.List(properties.Object(Variable), 'variables')
+    typ = properties.String('type', default="xor", deserializable=False)
+
+    def _attrs(self) -> List[str]:
+        return ["name", "headers", "variables", "typ"]
+
+    def __init__(self, *, name, headers, variables):
```

Comment:
As a dodge, can we put defaults in here?  Or would overlapping names collide in later processing?  If `name` and `headers` are no-op arguments, we could bypass the need for people to do bookkeeping.

Reply:
Are you asking if we can derive `name` and `headers` from `variables`?

Reply:
Sorry, I had this backwards in my brain, and we thinking we could avoid the issue where the `variables` we are feeding in required names and headers for no good reason.  Ignore.